lib/dash/dash_parser.js

```diff
@@ -2751,25 +2751,49 @@ shaka.dash.DashParser = class {
           return intValue;
         }
 
+        case 'tag:dolby.com,2015:dash:audio_channel_configuration:2015': {
+          // ETSI TS 103 190-2 v1.2.1, Annex G.3
+          const channelCountMapping =
+            [2, 1, 2, 2, 2, 2, 1, 2, 2, 1, 1, 1, 1, 2, 1, 1, 2, 2];
+          const hexValue = parseInt(value, 16);
+          if (!hexValue) {  // 0 or NaN
+            shaka.log.warning('Channel parsing failure! ' +
+                          'Ignoring scheme and value', scheme, value);
+            continue;
+          }
+          let numBits = 0;
+          for (let i = 0; i < channelCountMapping.length; i++) {
+            if (hexValue & (1<<i)) {
+              numBits += channelCountMapping[i];
+            }
+          }
+          if (numBits) {
+            return numBits;
+          }
+          continue;
+        }
+
         case 'tag:dolby.com,2014:dash:audio_channel_configuration:2011':
         case 'urn:dolby:dash:audio_channel_configuration:2011': {
-          // A hex-encoded 16-bit integer, in which each bit represents a
-          // channel.
-          let hexValue = parseInt(value, 16);
+          // Defined by https://ott.dolby.com/OnDelKits/DDP/Dolby_Digital_Plus_Online_Delivery_Kit_v1.5/Documentation/Content_Creation/SDM/help_files/topics/ddp_mpeg_dash_c_mpd_auchlconfig.html
+          const channelCountMapping =
+            [1, 1, 1, 1, 1, 2, 2, 1, 1, 2, 2, 2, 1, 2, 1, 1].reverse();
```

Comment:
Why write an array of literals and then ask for it to be reversed at runtime?

Reply:
The static list is in same order as the spec.  The list is reversed to make the code consistent with the code of the other identifiers.  I can redo to your preference.

Reply:
It's fine if you add a comment to explain.  Something similar to what you just said is fine.